### PR TITLE
Add null safety switch

### PR DESCRIPTION
**Pull request by johnpryan**
This changes the backend to the one used for https://nullsafety.dartpad.dev/

- [x] Add this feature to inject script
- [x] Update samples dropdown links when null safety mode is selected (create new gists)

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by johnpryan** [comment]
Need to get this working with the inject script

**Comment by johnpryan** [comment]
@RedBrogdon @domesticmouse this is ready for another look

File: lib/embed.dart [hunk 1]
```diff
@@ -761,7 +810,8 @@ class Embed {
     }
     tabController.setTabVisibility(
         'test', context.testMethod.isNotEmpty && _showTestCode);
-    menuButton.toggleAttr('hidden', context.testMethod.isEmpty);
+    // menuButton.toggleAttr('hidden', context.testMethod.isEmpty);
```
Comment (domesticmouse) [inline lib/embed.dart:813]: Is this left over from debugging something? Otherwise, please delete comment.

**Review by domesticmouse — APPROVED** [review]

File: lib/embed.dart (hunk 1 above)
Comment (johnpryan) [reply lib/embed.dart]: Oops - I'll remove this line. The menu used to hide if there wasn't a reason to show it, so now that the null safety option is there it means that we should always keep it visible.

**Comment by johnpryan** [comment]
I noticed that completions aren't returning from the null safety version of dart_services: https://github.com/dart-lang/dart-pad/issues/1660

File: lib/util/query_params.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import 'dart:html';
```
Comment (RedBrogdon) [inline lib/util/query_params.dart:1]: Out of scope for this PR, but it would be good to get the rest of the query parameters in here as well.

**Review by RedBrogdon — APPROVED** [review]

File: lib/util/query_params.dart (hunk 2 above)
Comment (johnpryan) [reply lib/util/query_params.dart:1]: You're right, created an issue to track: https://github.com/dart-lang/dart-pad/issues/1664

**Pull request merged**